### PR TITLE
refactoring equals

### PR DESCRIPTION
**Pull request by GerardPaligot**

### CODE REVIEW

File: src/main/java/spoon/support/visitor/equals/EqualsCheckerInheritance.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+			isNotEqual = true;
+			return;
+		}
+		super.scanCtStatement(s);
+	}
+
+	@Override
+	public void scanCtModifiable(CtModifiable m) {
+		final CtModifiable peek = (CtModifiable) this.other;
+/*
```
Comment (monperrus) [inline src/main/java/spoon/support/visitor/equals/EqualsCheckerInheritance.java]: add them again?

File: src/main/java/spoon/support/visitor/equals/EqualsCheckerInheritance.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.CtModifiable;
+import spoon.reflect.declaration.CtNamedElement;
+import spoon.reflect.declaration.CtParameter;
+import spoon.reflect.reference.CtArrayTypeReference;
+import spoon.reflect.reference.CtExecutableReference;
+import spoon.reflect.reference.CtReference;
+import spoon.reflect.visitor.CtInheritanceScanner;
+
+public class EqualsCheckerInheritance extends CtInheritanceScanner {
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/visitor/equals/EqualsCheckerInheritance.java]: EqualsChecker

File: src/main/java/spoon/generating/equals/EqualsVisitor.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.visitor.CtBiScanner;
+import spoon.support.visitor.equals.EqualsCheckerInheritance;
+
+/**
+ * Used to check equality between an element and another one.
+ *
+ * This class is generated automatically by the processor {@link spoon.generating.EqualsVisitorGenerator}.
+ */
+class EqualsVisitor extends CtBiScanner {
```
Comment (GerardPaligot) [inline src/main/java/spoon/generating/equals/EqualsVisitor.java]: EqualsVisitorTemplate

File: src/main/java/spoon/generating/scanner/CtBiScanner.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+/**
+ * This visitor implements a deep-search scan on the model for 2 elements.
+ *
+ * Ensures that all children nodes are visited once, a visit means three method
+ * calls, one call to "enter", one call to "exit" and one call to biScan.
+ *
+ * Is used by EqualsVisitor.
+ */
+abstract class CtBiScanner implements CtVisitor {
```
Comment (GerardPaligot) [inline src/main/java/spoon/generating/scanner/CtBiScanner.java]: BiScannerTemplate

File: src/main/java/spoon/generating/scanner/CtBiScannerScanner.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.code.CtLocalVariable;
+import spoon.reflect.code.CtVariableAccess;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.reference.CtTypeReference;
+import spoon.reflect.visitor.CtScanner;
+
+public class CtBiScannerScanner extends CtScanner {
```
Comment (GerardPaligot) [inline src/main/java/spoon/generating/scanner/CtBiScannerScanner.java]: anonymous class?

File: src/main/java/spoon/reflect/visitor/CtBiScanner.java [hunk 6]
```diff
@@ -0,0 +1,865 @@
...
+
+package spoon.reflect.visitor;
+
+
+/**
+ * This visitor implements a deep-search scan on the model for 2 elements.
+ *
+ * Ensures that all children nodes are visited once, a visit means three method
+ * calls, one call to "enter", one call to "exit" and one call to biScan.
+ *
```
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/CtBiScanner.java]: mega javadoc GENERATED DO NOT MODIFY

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Pull request merged**